File: src/main/java/infinitespire/patches/SavePatch.java
package infinitespire.patches;

import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.saveAndContinue.SaveFile;
import infinitespire.InfiniteSpire;

public class SavePatch {
	@SpirePatch(cls = "com.megacrit.cardcrawl.saveAndContinue.SaveAndContinue", method = "loadSaveString",
			paramtypes= {"java.lang.String"})
	public static class LoadGame1 {

		public static void Prefix(String filePath) {
			InfiniteSpire.loadData();
		}
	}

	@SpirePatch(cls = "com.megacrit.cardcrawl.saveAndContinue.SaveAndContinue", method = "loadSaveFile",
			paramtypes= {"java.lang.String"})
	public static class LoadGame2 {
		public static void Prefix(String filePath) {
			InfiniteSpire.loadData();
		}
	}
	
	@SpirePatch(cls = "com.megacrit.cardcrawl.saveAndContinue.SaveAndContinue", method = "save")
	public static class SaveGame {
		
		public static void Prefix(SaveFile save) {
			InfiniteSpire.saveData();
		}
	}
	
	@SpirePatch(cls = "com.megacrit.cardcrawl.saveAndContinue.SaveAndContinue", method = "deleteSave")
	public static class DeleteSave {
		public static void Prefix(AbstractPlayer p) {
			InfiniteSpire.clearData();
		}
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/patches/io/SavePatch.kt
package com.blanktheevil.infinitespire.patches.io

import com.blanktheevil.infinitespire.InfiniteSpire
import com.blanktheevil.infinitespire.extensions.log
import com.blanktheevil.infinitespire.interfaces.Savable
import com.evacipated.cardcrawl.modthespire.lib.SpirePatch
import com.evacipated.cardcrawl.modthespire.lib.SpirePostfixPatch
import com.megacrit.cardcrawl.saveAndContinue.SaveFile

@Suppress("unused", "UNUSED_PARAMETER")
@SpirePatch(
  clz = SaveFile::class,
  method = SpirePatch.CONSTRUCTOR,
  paramtypez = [SaveFile.SaveType::class]
)
object SavePatch {
  @JvmStatic
  @SpirePostfixPatch
  fun saveInfiniteSpireData(saveFile: SaveFile) {
    log.info("Saving Data...")
    Savable.savables.forEach {
      it.beforeConfigSave(InfiniteSpire.saveData)
    }
    InfiniteSpire.saveData.save()
  }
}